frontend/packages/multicluster-sdk/src/internal/fleetK8sWatchResourceStore.ts

```diff
@@ -0,0 +1,309 @@
+/* Copyright Contributors to the Open Cluster Management project */
+import { create } from 'zustand'
+import { subscribeWithSelector } from 'zustand/middleware'
+
+export interface CacheEntry<T = any> {
+  data: T
+  loaded: boolean
+  error?: any
+  timestamp: number
+  lastAccessed: number
+}
+
+export interface SocketCacheEntry {
+  socket: WebSocket
+  timestamp: number
+  lastAccessed: number
+  refCount: number // Track how many hooks are using this socket
+}
+
+export interface FleetK8sWatchResourceStore {
+  // Resource cache
+  resourceCache: Record<string, CacheEntry>
+
+  // Socket cache
+  socketCache: Record<string, SocketCacheEntry>
+
+  // Cache TTL in milliseconds (default 5 minutes)
+  cacheTTL: number
```

Comment:
TTL caching seemed standard practice for this, but am open to feedback on implementation and time limit used.